bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/handler/OpenWeatherMapWeatherAndForecastHandler.java

```diff
@@ -82,8 +83,11 @@ public OpenWeatherMapWeatherAndForecastHandler(Thing thing) {
     }
 
     @Override
-    public void initialize() {
-        super.initialize();
+    protected void initializeThing(@Nullable ThingHandler bridgeHandler, @Nullable ThingStatus bridgeStatus) {
+        super.initializeThing(bridgeHandler, bridgeStatus);
+        if (bridgeHandler == null || bridgeStatus != ThingStatus.ONLINE) {
+            return;
+        }
```

Comment:
You should eliminate the requirement for subclasses to have to call super and then exit early. Instead you should just move all the  `AbstractOpenWeatherMapHandler.initializeThing` logic into `AbstractOpenWeatherMapHandler.initialize`, make `AbstractOpenWeatherMapHandler.initializeThing` abstract and only call `initializeThing` when `bridgeHandler != null && bridgeStatus == ThingStatus.ONLINE`.

Reply:
I see what you mean and you're right. I am going to improve that.